includes/class-utils.php

```diff
@@ -44,4 +44,47 @@ function ( $matches ) {
 		return $parts;
 	}
 
+	/**
+	 * Takes a request URL and "normalises" it, stripping common elements.
+	 * Removes scheme and host from the URL, as redirects should be independent of these.
+	 *
+	 * @param string $url URL to transform.
+	 * @return string|WP_Error Transformed URL; error if validation failed.
+	 */
+	public static function normalise_url( $url ) {
```

Comment:
Moving this method isn't needed for this PR.

Reply:
I'd like to a point where this behaviour is extracted into it's own service class:

```php
interface Url_Normalizer {
	public function normalize( $url );
}

class Default_Url_Normalizer implements Url_Normalizer {
	public function normalize( $url ) {
		...
	}
}
```

Then we can have a URL class that can accept an `Url_Normalizer`, and be passed an instance of `Default_Url_Normalizer` for normal running, or a mock object for testing.

This concrete implementation could have a filter wrapped around it, so if customers want to provide their own concrete implementation that handled normalizing in a different way, they could.

Reply:
Reverted code change and added a new Unit test class `RedirectsTests` to keep the unit tests for `normalize_url`